src/pages/index.js

```diff
@@ -3,141 +3,201 @@ import "../pages/index.css";
   const cardElement = renderCard(item);
-  cardSection.addItem(cardElement.getCard());
+  cardSection.addItem(cardElement);
 }
 
 //RENDER CARD
 function renderCard(data) {
   const card = new Card(data, "#card-template", {
     handleImageClick: () => imagePopup.open(data),
+    handleDeleteClick,
+    handleCardLike
   });
-  return card;
+  return card.getCard();
 }
 
 //IMAGE POPUP
 const imagePopup = new PopupWithImage("#preview-image-modal");
 imagePopup.setEventListeners();
 
-//FORM POPUP
+//DELETE CARD POPUP
+const deleteCardPopup = new PopupWithConfirmation("#card__delete-modal");
+deleteCardPopup.setEventListeners();
+
+
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.setSubmitText(true, "Deleting...");
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        deleteCardPopup.close();
+        card.removeCard();
+      })
+      .catch((err) => console.log(err))
+      .finally(() => deleteCardPopup.setSubmitText(false));
+  });
+}
+
 
+
+//FORM POPUP
 const openImagePopup = new PopupWithForm(
   "#add-new-card",
   handleImageFormSubmit
 );
-const editProfile = new PopupWithForm(
-  "#profile-edit-modal",
-  handleFormSubmit
+const editProfile = new PopupWithForm("#profile-edit-modal", handleFormSubmit);
+const profilePicEdit = new PopupWithForm(
+  "#profile__image-edit",
+  profilePicEditSubmit
 );
+
 editProfile.setEventListeners();
+profilePicEdit.setEventListeners();
 openImagePopup.setEventListeners();
 
-const editFormValidator = new FormValidator(config, profileModalForm);
-const cardFormValidator = new FormValidator(config, addNewCardModal);
+const handleEditFormSubmit = new FormValidator(config, profileModalForm);
+const handleProfileFormSubmit = new FormValidator(config, profilePicModal);
+const handleCardFormSubmit = new FormValidator(config, addNewCardModal);
 
-editFormValidator.enableValidation();
-cardFormValidator.enableValidation();
+handleEditFormSubmit.enableValidation();
+handleProfileFormSubmit.enableValidation();
+handleCardFormSubmit.enableValidation();
 
 //FORM SUBMIT
 function handleFormSubmit(data) {
-  console.log(data)
-  userInfo.setUserInfo(data)
-  editProfile.close();
+  editProfile.setSubmitText(true, "Saving...");
+  api
+    .profileUpdate(data)
+    .then(() => {
+      userInfo.setUserInfo(data);
+      editProfile.setSubmitText(false);
+      editProfile.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+
 }
 
 // //IMAGE SUBMIT
-function handleImageFormSubmit() {
-  const name = newCardModalTitle.value;
-  const link = newCardModalUrl.value;
-  generateCard({ name, link });
-  openImagePopup.close();
+function handleImageFormSubmit(data) {
+  const name = data.name;
+  const link = data.url;
+  openImagePopup.setSubmitText(true, "Saving...");
+  api
+    .addCard(name, link)
+    .then((card) => {
+      generateCard(card);
+      openImagePopup.setSubmitText(false)
```

Comment:
* You haven't corrected my remark https://github.com/MrSurge87/se_project_aroundtheus/pull/1/files#r1389001567  You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.